Sample develop

Zbigniew-Jedlinski-Mobica opened the pull request:
## Description

Please include a summary of the change, new sample or fixed issue. Please also include relevant motivation and context.
Please read the [contribution guidelines](https://github.com/KhronosGroup/Vulkan-Samples/tree/master/CONTRIBUTING.md)

Fixes #<issue number>

## General Checklist:

Please ensure the following points are checked:

- [ ] My code follows the [coding style](https://github.com/KhronosGroup/Vulkan-Samples/tree/master/CONTRIBUTING.md#Code-Style)
- [ ] I have reviewed file [licenses](https://github.com/KhronosGroup/Vulkan-Samples/tree/master/CONTRIBUTING.md#Copyright-Notice-and-License-Template)
- [ ] I have updated the header Copyright to reflect the current year (CI build will fail if Copyright is out of date)
- [ ] I have commented any added functions (in line with Doxygen)
- [ ] I have commented any code that could be hard to understand
- [ ] My changes do not add any new compiler warnings
- [ ] My changes do not add any new validation layer errors or warnings
- [ ] I have used existing framework/helper functions where possible
- [ ] My changes build on Windows, Linux, macOS and Android. Otherwise I have [documented any exceptions](https://github.com/KhronosGroup/Vulkan-Samples/tree/master/CONTRIBUTING.md#General-Requirements)
- [ ] My changes do not add any regressions
- [ ] I have tested every sample to ensure everything runs correctly
- [ ] This PR describes the scope and expected impact of the changes I am making

## Sample Checklist

If your PR contains a new or modified sample, these further checks must be carried out *in addition* to the General Checklist:
- [ ] I have tested the sample on at least one compliant Vulkan implementation
- [ ] If the sample is vendor-specific, I have [tagged it appropriately](https://github.com/KhronosGroup/Vulkan-Samples/tree/master/CONTRIBUTING.md#General-Requirements)
- [ ] I have stated on what implementation the sample has been tested so that others can test on different implementations and platforms
- [ ] Any dependent assets have been merged and published in downstream modules
- [ ] For new samples, I have added a paragraph with a summary to the appropriate chapter in the [samples readme](./../samples/README.md)

Zbigniew-Jedlinski-Mobica committed:
Start patch_control_points sample

Zbigniew-Jedlinski-Mobica committed:
Prepare patch control points draft

Karina-Murawko-Wisniewska-Mobica committed:
Remove unused code

Karina-Murawko-Wisniewska-Mobica committed:
Remove baseline related code

Karina-Murawko-Wisniewska-Mobica committed:
Remove push constant from tese shader

Karina-Murawko-Wisniewska-Mobica committed:
Add push constant for model

Zbigniew-Jedlinski-Mobica committed:
Update sample - two terrain models

Created two terrain models

Karina-Murawko-Wisniewska-Mobica committed:
Add README draft

Karina-Murawko-Wisniewska-Mobica committed:
Remove method declaration

Karina-Murawko-Wisniewska-Mobica committed:
Final refactor

Zbigniew-Jedlinski-Mobica committed:
Modification way of tessellation in shader

Updation camera setting and control shader file

Zbigniew-Jedlinski-Mobica committed:
Update tess.tesc

Zbigniew-Jedlinski-Mobica committed:
Add screenshot

Update readme and add screenshot

Karina-Murawko-Wisniewska-Mobica commented on samples/extensions/patch_control_points/README.md:
```diff
@@ -0,0 +1,146 @@
+-->
+
+# Patch control point 
+
```
```suggestion

![Sample](./images/patch_control_point_screenshot.png)

```

Karina-Murawko-Wisniewska-Mobica replied:
The screenshot link in readme is missing.

Michal-Masiukiewicz-Mobica commented on shaders/patch_control_points/tess.tese:
```diff
@@ -0,0 +1,51 @@
+	return vec3(gl_TessCoord.x) * v0 + vec3(gl_TessCoord.y) * v1 + vec3(gl_TessCoord.z) * v2;
+}
+
+vec4 interpolate3D(vec4 v0, vec4 v1, vec4 v2)
```
Seems like only this interpolate3D function is required. Maybe remove others?

Michal-Masiukiewicz-Mobica commented on shaders/patch_control_points/tess.tesc:
```diff
@@ -0,0 +1,98 @@
+
+void main()
+{
+	if (gl_InvocationID == 0)
```
Maybe some comment why (gl_InvocationID == 0) is special?
I think for somebody who doesn't know how tesselation shaders work (me) it would be helpful.

Zbigniew-Jedlinski-Mobica committed:
Update README.md

Zbigniew-Jedlinski-Mobica committed:
Removed gl_InvocationID condition

Zbigniew-Jedlinski-Mobica committed:
Update tess.tese

Szymon-Rajczakowski commented on samples/extensions/patch_control_points/patch_control_points.cpp:
```diff
@@ -0,0 +1,620 @@
+
+#include <memory>
+#include <string>
+#include <vector>
```
These includes are not needed because are included indirectly by #include "patch_control_points.h"

Szymon-Rajczakowski commented on samples/extensions/patch_control_points/patch_control_points.cpp:
```diff
@@ -0,0 +1,620 @@
+
+PatchControlPoints::PatchControlPoints()
+{
+	title = "Extended Dynamic State2";
```
Change title to Patch Control Points or sth like that

Szymon-Rajczakowski commented on samples/extensions/patch_control_points/patch_control_points.cpp:
```diff
@@ -0,0 +1,620 @@
+	dynamic_state.pDynamicStates    = dynamic_state_enables.data();
+	dynamic_state.dynamicStateCount = static_cast<uint32_t>(dynamic_state_enables.size());
+
+	VK_CHECK(vkCreateGraphicsPipelines(get_device().get_handle(), VK_NULL_HANDLE, 1, &graphics_create, VK_NULL_HANDLE, &pipeline.dynamically_tessellation));
```
why don't you use pipeline_cache here?

Szymon-Rajczakowski commented on samples/extensions/patch_control_points/patch_control_points.cpp:
```diff
@@ -0,0 +1,620 @@
+
+		draw_model(model, draw_cmd_buffer);
+
+		//	dynamiclly tessellation
```
typo

Szymon-Rajczakowski commented on samples/extensions/patch_control_points/patch_control_points.cpp:
```diff
@@ -0,0 +1,620 @@
+
+/**
+ * 	@fn void PatchControlPoints::create_descriptor_sets()
+ * 	@brief Creating descriptor sets for 3 separate pipelines.
```
2 separate pipelines

Zbigniew-Jedlinski-Mobica committed:
Updating as suggested in pull request

Removed unnecessary include files
The title of the program has been changed
Added pipeline_cache to vkCreateGraphicsPipelines
Fixed typos

Zbigniew-Jedlinski-Mobica committed:
removing unnecessary include files

The include files are included in other files
